Adamant/Services/AdamantCoinStorageService.swift

```diff
@@ -104,24 +107,49 @@ private extension AdamantCoinStorageService {
         )
         .sink { [weak self] notification in
             let changes = notification.managedObjectContextChanges(of: CoinTransaction.self)
-
+            
+            guard self != nil, self?.coinId != nil && self?.coinAddress != nil else {
```

Comment:
guard let self else { return } would eliminate the need to force-uwrap self later